components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1683,6 +1684,32 @@ void AdsServiceImpl::OnPrefsChanged(
     } else {
       Stop();
     }
+
+    // Record P3A.
+    using brave_rewards::AdsP3AState;
```

Comment:
I think this code should be encapsulated in a helper method in rewards_p3a.h

Reply:
it also needs tests. The data we get isn't going to be very useful if the values are not set correctly

Reply:
fuxed